### DISCUSSION TITLE
Add support for dnxhd codec

### DISCUSSION THREAD

**Discussion by jaspreet-dimension** [AcademySoftwareFoundation/OpenRV#490] [Q&A] [answered]
Hi everyone,

I'm new to Open RV and managed to build it last week. I need to add support for the dnxhd video codec. Does anyone know how I can go about doing this? In Chapter 15 of the User Manual, it says that I might need to add a plugin. Is there a plugin available for dnxhd? I haven't managed to find anything yet.

Thanks

**Comment by bernie-laberge** [comment, marked as answer]
Hello @jaspreet-dimension ,

Since that it is a non free codecs, it has been disabled by default in Open RV.
https://github.com/AcademySoftwareFoundation/OpenRV/blob/a2bfba4ad37714cdb4a0b7f5e5e30a6a05e5df8d/cmake/dependencies/ffmpeg.cmake#L206

I hope this helps

**Reply by jaspreet-dimension** [reply]
Thanks! I tried removing the lines for the decoder/encoder but looked like it didn't work. Is there anything else I need to do?

**Comment by bernie-laberge** [comment]
You might have to delete your _build directory and rebuild Open RV. 
I suspect that there is a dependency missing in the Open RV build system with respect to FFmpeg (FFmpeg doesn't get replaced in the stage area if it gets modified). 
Sorry about that.

**Reply by jaspreet-dimension** [reply]
Still no luck unfortunately, I deleted the _build and _install folders and reran rvbootstrap and rvinst and still nothing... Attached my `rv.exe -formats` output here in case that helps

[formats3.txt](https://github.com/user-attachments/files/15874019/formats3.txt)

**Reply by bernie-laberge** [reply]
OK I'm rebuilding my Open RV right now without the line above. 
I will test it with this file once it finishes building: https://cloud.slomo.tv/s/r7m4LAC99HFmWx3
I'll keep you posted once my build completes.

**Comment by jaspreet-dimension** [comment]
Those files seem to work fine on my latest build of rv.exe, they're audio
only files right?

How come dnxhd doesn't show up when I do rv.exe -formats?

**Reply by bernie-laberge** [reply]
You are right, there are a couple of audio files but one of them is a video file: Pgm1_HD_DNX120_25p_0_tc_12_23_33_12_v1.mxf.

I believe that dnxhd doesn't show up when doing rv.exe -formats, because it looks like the -formats option only lists the file extensions (.mxf) and not the actually codecs are are enabled in FFmpeg. Something that could be improved I guess.

**Reply by jaspreet-dimension** [reply]
Thank you anyway, you were a great help!
